src/app/clusters/operational-credentials/operational-credentials.cpp

```diff
@@ -33,33 +36,193 @@
 
 using namespace chip;
+using namespace ::chip::DeviceLayer;
+using namespace ::chip::Transport;
+
+EmberAfStatus writeFabricAttribute(uint8_t endpoint, AttributeId attributeId, uint8_t * buffer, int32_t index = -1)
+{
+    EmberAfAttributeSearchRecord record;
+    record.endpoint         = endpoint;
+    record.clusterId        = ZCL_OPERATIONAL_CREDENTIALS_CLUSTER_ID;
+    record.clusterMask      = CLUSTER_MASK_SERVER;
+    record.manufacturerCode = EMBER_AF_NULL_MANUFACTURER_CODE;
+    record.attributeId      = attributeId;
+
+    // When reading or writing a List attribute the 'index' value could have 3 types of values:
+    //  -1: Read/Write the whole list content, including the number of elements in the list
+    //   0: Read/Write the number of elements in the list, represented as a uint16_t
+    //   n: Read/Write the nth element of the list
+    //
+    // Since the first 2 bytes of the attribute are used to store the number of elements, elements indexing starts
+    // at 1. In order to hide this to the rest of the code of this file, the element index is incremented by 1 here.
+    // This also allows calling writeAttribute() with no index arg to mean "write the length".
+    
+    return emAfReadOrWriteAttribute(&record, NULL, buffer, 0, true, index + 1);
+}
+
+EmberAfStatus writeFabric(FabricId fabricId, NodeId nodeId, uint16_t vendorId, int32_t index)
+{
+    EmberAfStatus status    = EMBER_ZCL_STATUS_SUCCESS;
+    AttributeId attributeId = ZCL_FABRICS_ATTRIBUTE_ID;
+
+    EmberAfFabricDescriptor fabricDescriptor;
+    fabricDescriptor.FabricId = fabricId;
+    fabricDescriptor.NodeId   = nodeId;
+    fabricDescriptor.VendorId = vendorId;
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Writing admin into attribute store at index %d: fabricId %" PRIX64 
+                            ", nodeId %" PRIX64 " vendorId %" PRIX16, index, fabricId, nodeId, vendorId);
+    status =  writeFabricAttribute(0, attributeId, (uint8_t *) &fabricDescriptor, index);
+    return status;
+}
+
+CHIP_ERROR writeAdminsIntoFabricsListAttribute(void)
+{
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Call to writeAdminsIntoFabricsListAttribute");
+
+    // Loop through admins
+    auto pairing = GetGlobalAdminPairingTable().cbegin();
+    int32_t fabricIndex = 0;
+    while (pairing != GetGlobalAdminPairingTable().cend())
+    { 
+        NodeId nodeId = pairing->GetNodeId();
+        uint64_t fabricId = pairing->GetFabricId();
+        uint16_t vendorId = pairing->GetVendorId();
+
+        // Skip over uninitialized admins
+        if (nodeId != kUndefinedNodeId && fabricId != kUndefinedFabricId && vendorId != kUndefinedVendorId)
+        {
+            if (writeFabric(fabricId, nodeId, vendorId, fabricIndex) != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to write admin with fabricId %" PRIX64 " in fabrics list", fabricId);
+            }
+            fabricIndex ++;
```

Comment:
Should we really increment the index if the write failed?

Reply:
I don't see why you'd want to leave the user with a partial list. I'd say keep going and have just one missing (not sure why only one would fail).

Reply:
Well, if the write failed we don't have one "missing".  We have one that's set to whatever values happened to be in memory there already, so if something tries to read it, what gets read?  If we're lucky, all 0.  If we're not lucky, some random memory.

Reply:
The only think I can do here is stop the fabricCount from going up that way and break out of for loop, so that if we had stored 2 admins and 3rd one failed to write, we will stop, log the error. And store a count of 2 so that when the fabrics are read, only first 2 are returned. Could also choose to store a count of 0 if an error happens...